assets/src/edit-story/components/canvas/karma/carousel.karma.js

```diff
@@ -0,0 +1,220 @@
+    const storyContext = await fixture.renderHook(() => useStory());
+    return storyContext.state.selectedElementIds;
+  }
+
+  async function getPageIds() {
+    const {
+      state: { pages },
+    } = await fixture.renderHook(() => useStory());
+    return pages.map(({ id }) => id);
+  }
+
+  function getThumbnail(index) {
+    const thumbnails = carousel.querySelectorAll(
+      '[role="listbox"] button[role="option"]'
+    );
+    return thumbnails[index];
+  }
+
+  it('should select the current page', async () => {
+    expect(await getCurrentPageId()).toEqual(page1);
+    expect(await getSelection()).toEqual([element1.id]);
+  });
+
+  it('should click into carousel on the first page', async () => {
+    await fixture.events.mouse.clickOn(getThumbnail(0), 5, 5);
+    expect(await getCurrentPageId()).toEqual(page1);
+    expect(await getSelection()).toEqual([]);
+  });
+
+  it('should exit the carousel on Esc', async () => {
+    await fixture.events.mouse.clickOn(getThumbnail(0), 5, 5);
+    await waitFor(() => {
+      if (!carousel.contains(document.activeElement)) {
+        throw new Error('Focus is not set on the carousel yet');
+      }
+    });
+
+    // Exit.
+    await fixture.events.keyboard.press('Esc');
+    const framesLayer = fixture.querySelector('[data-testid="FramesLayer"]');
+    await waitFor(() => {
+      if (!framesLayer.contains(document.activeElement)) {
+        throw new Error('Focus is not set on the canvas yet');
+      }
+    });
+  });
+
+  it('should click into carousel on the second page', async () => {
+    await fixture.events.mouse.clickOn(getThumbnail(1), 5, 5);
+    expect(await getCurrentPageId()).toEqual(page2);
+    expect(await getSelection()).toEqual([]);
+  });
+
+  it('should navigate the page with keys', async () => {
+    await fixture.events.mouse.clickOn(getThumbnail(1), 5, 5);
+    expect(await getCurrentPageId()).toEqual(page2);
+
+    // Go left.
+    await fixture.events.keyboard.press('left');
+    expect(await getCurrentPageId()).toEqual(page1);
+
+    // Go left again: end of the line.
+    await fixture.events.keyboard.press('left');
+    expect(await getCurrentPageId()).toEqual(page1);
+
+    // Go right.
+    await fixture.events.keyboard.press('right');
+    expect(await getCurrentPageId()).toEqual(page2);
+
+    // Go right again.
+    await fixture.events.keyboard.press('right');
+    expect(await getCurrentPageId()).toEqual(page3);
+
+    // Go right again.
+    await fixture.events.keyboard.press('right');
+    expect(await getCurrentPageId()).toEqual(page4);
+
+    // Go right again: end of the line.
+    await fixture.events.keyboard.press('right');
+    expect(await getCurrentPageId()).toEqual(page4);
+  });
+
+  it('should reorder the page with keys', async () => {
+    await fixture.events.mouse.clickOn(getThumbnail(1), 5, 5);
+    expect(await getCurrentPageId()).toEqual(page2);
+
+    // Order left.
+    await fixture.events.keyboard.shortcut('mod+left');
```

Comment:
Note: currently in the compact view, it's not possible to reorder pages from carousel using mouse, however, now it's possible via keyboard. I guess that's fine there's a difference?

Reply:
Probably not very meaningful. I filed https://github.com/google/web-stories-wp/issues/2575 to clarify.